ServiceBusDashboard/Controllers/DashboardController.cs

```diff
@@ -0,0 +1,187 @@
+
+                replaceTheseStartWordsWithDots
+                    .Split(',')
+                    .ForEach(r => name = name.StartsWith(r) ? name.Replace(replaceTheseStartWordsWithDots, "...") : name);
+
+                return name;
+            }
+
+            var queues = namespaceManager.GetQueues();
+            var topics = namespaceManager.GetTopics();
+
+            var data = new DashboardModel()
+            {
+                ConnectionString= sbConnectionString,
+                Queues = queues
+                    .Where(x => !IsQueueIgnored(x.Path))
+                    .Select(x => new SbQueue()
+                    {
+                        Name = ReplaceTheseStartWordsWithDots(x.Path),
+                        ActiveMessageCount = x.MessageCount,
+                        DeadLetterMessageCount = x.MessageCountDetails.DeadLetterMessageCount
+                    })
+                    .ToArray(),
+                Topics = topics
+                    .Where(x => !IsTopicIgnored(x.Path))
+                    .Select(x => new SbTopic()
+                    {
+                        Name = ReplaceTheseStartWordsWithDots(x.Path),
+                        ActiveMessageCount = x.MessageCountDetails.ActiveMessageCount,
+                        DeadLetterMessageCount = x.MessageCountDetails.DeadLetterMessageCount,
+                        Subscriptions = namespaceManager.GetSubscriptions(x.Path)
+                            .Where(xx => !IsSubscriptionIgnored(xx.Name))
+                            .Select(xx => new SbSubscription()
+                            {
+                                Name = ReplaceTheseStartWordsWithDots(xx.Name),
+                                ActiveMessageCount = xx.MessageCount,
+                                DeadLetterMessageCount = xx.MessageCountDetails.DeadLetterMessageCount
+                            })
+                        .ToArray()
+                    })
+                    .ToArray()
+            };
+
+            var excludedSubscriptions = excludeSubscriptions.Split(',').Select(x => x.Split('/')).Select(x => new { Topic = x[0], Subscription = x[1] });
+            data.Topics.ForEach(x => x.Subscriptions.ForEach(xx =>
+              {
+                  if (excludedSubscriptions.Any(xxx => xxx.Topic == x.Name && xxx.Subscription == xx.Name))
+                      xx.Excluded = true;
+              }));
+
+            data.Sort();
+
+            var body = RazorUtil.RenderViewToString("Dashboard", "~/Views/Home/DashboardView.cshtml", data);
+
+            return Ok(body);
+        }
+    }
+
+    public class DashboardModel
+    {
+        public SbConnectionString ConnectionString { get; set; }
+        public SbQueue[] Queues { get; set; }
+        public SbTopic[] Topics { get; set; }
+
+        public void Sort()
+        {
+            Array.Sort(Queues);
+            Array.Sort(Topics);
+            Topics.ForEach(x => Array.Sort(x.Subscriptions));
+        }
+    }
+
```

Comment:
Can also move these classes out of the controller

Reply:
done